Documentation/Evolution/RegexLiteralPitch.md

```diff
@@ -0,0 +1,259 @@
+
+```swift
+SomeBuilder {
+  SomeType()
+  { print("hello") }
+  AnotherType()
+}
+```
+
+`{ print("hello") }` will be parsed as a trailing closure to `SomeType()` rather than as a separate element to the result builder.
+
+
+## Future Directions
+
+### Typed captures
+
+Typed captures would statically represent how many captures and of what kind are present in a regex literals. They could produce a `Substring` for a regular capture, `Substring?` for a zero-or-one capture, and  `Array<Substring>` (or a lazy collection) for a zero(or one)-or-more capture. These are worth exploring, especially in the context of the [start of variadic generics][variadics] support, but we'd like to keep this pitch and discussion focused to the details presented.
+
+### Other regex literals
+
+Multi-line extensions to regex literals is considered future work. Generally, we'd like to encourage refactoring into `Pattern` when the regex gets to that degree of complexity.
+
+User-specified [choice of quote delimiters][perlquotes] is considered future work. A related approach to this could be a "raw" regex literal analogous to [raw strings][rawstrings]. For example (total strawperson), an approach where `n` `#`s before the opening delimiter would requires `n` `#` at the end of the trailing delimiter as well as requiring `n-1` `#`s to access metacharacters.
+
+```txt
+// All of the below are trying to match a path like "/tmp/foo/bar/File.app/file.txt"
+
+/\/tmp\/.*\/File\.app\/file\.txt/
+#//tmp/.*/File\.app/file\.txt/#
+##//tmp/#.#*/File.app/file.txt/##
+```
+
+"Swiftier" literals, such as with non-semantic whitespace (e.g. [Raku's][rakuregex]), is future work. We'd want to strongly consider using a different backing technology for Swifty matching literals, such as PEGs.
+
+Fully-custom literal support, that is literals whose bodies are not parsed and there is no default type available, is orthogonal to this work. It would require support for compilation-time Swift libraries in addition to Swift APIs for the compiler and type system.
+
+
+### Further extension to Swift language constructs
+
+Other language constructs, such as raw-valued enums, might benefit from further regex enhancements.
+
+```swift
+enum CalculatorToken: Regex {
+  case wholeNumber = /\d+/
+  case identifier = /\w+/
+  case symbol = /\p{Math}/
+  ...
+}
+```
+
+As mentioned in the overview, general purpose extensions to Swift (syntactic) pattern matching could benefit regex
+
+```swift
+func parseField(_ field: String) -> ParsedField {
+  switch field {
+  case let text <- /#\s?(.*)/:
+    return .comment(text)
+  case let (l, u) <- /([0-9A-F]+)(?:\.\.([0-9A-F]+))?/:
+    return .scalars(Unicode.Scalar(hex: l) ... Unicode.Scalar(hex: u ?? l))
+  case let prop <- GraphemeBreakProperty.init:
+    return .property(prop)
+  }
+}
+```
+
+We'd like more community feedback on these future directions.
```

Comment:
Do we actually want community feedback on these directions in this thread? Seem like that'd be better left for the overview thread.